src/view/segmentation.js

```diff
@@ -0,0 +1,82 @@
+/* globals newspack_popups_view */
+
+import {
+	debug,
+	closeOverlay,
+	getBestPrioritySegment,
+	getIntersectionObserver,
+	handleSeen,
+	logPageview,
+	shouldPromptBeDisplayed,
+} from './utils';
+
+/**
+ * Match reader to segments.
+ */
+export const handleSegmentation = prompts => {
+	const segments = newspack_popups_view?.segments || {};
+	const matchingSegment = getBestPrioritySegment( segments );
+	debug( 'matchingSegment', matchingSegment );
+
+	window.newspackRAS = window.newspackRAS || [];
+	window.newspackRAS.push( ras => {
+		// Log a pageview for frequency counts.
+		logPageview( ras );
+		let overlayDisplayed;
+
+		prompts.forEach( prompt => {
+			const promptId = prompt.getAttribute( 'id' );
+			const isOverlay = prompt.classList.contains( 'newspack-lightbox' );
+
+			// Attach event listners to overlay close buttons.
+			const closeButtons = [
+				...prompt.querySelectorAll( '.newspack-lightbox__close, button.newspack-lightbox-overlay' ),
+			];
+			closeButtons.forEach( closeButton => {
+				closeButton.addEventListener( 'click', closeOverlay );
+			} );
+
+			// Check segmentation.
+			const shouldDisplay = shouldPromptBeDisplayed(
+				prompt,
+				matchingSegment,
+				ras,
+				overlayDisplayed
+			);
```

Comment:
I'm getting this returning `true` for every overlay prompt on the homepage, which doesn't render all the prompts on the page but dispatches the `prompt_seen` activity. Here's the localStorage on a fresh session with RAS defaults:

<img width="562" alt="image" src="https://github.com/Automattic/newspack-popups/assets/820752/36564c1b-d54e-4e5b-a8c0-41079bb9d298">

It's unclear to me why the `overlayDisplayed` is being sent as the `override` argument in the function. This causes every`shouldPromptBeDisplayed` to be forced as `true`.

Reply:
Whoops, that was a late-day logic mistake. 🤦 This should fix it: dd3fbd3